src/index/CompressedBlockPrefiltering.cpp

```diff
@@ -0,0 +1,170 @@
+    return std::vector<BlockMetadata>(input.begin(),
+                                      lowerIndex(input, evaluationColumn));
+  } else if constexpr (Comparison == GE || Comparison == GT) {
+    // Complementary block selection to LT and LE
+    return std::vector<BlockMetadata>(lowerIndex(input, evaluationColumn),
+                                      input.end());
+  } else if constexpr (Comparison == EQ) {
+    return std::vector<BlockMetadata>(lowerIndex(input, evaluationColumn),
+                                      upperIndex(input, evaluationColumn));
+  } else {
+    // CompOP::NE
+    // Get the indices that define the boundaries w.r.t. blocks potentially
+    // containing non-equal values.
+    auto firstBound = lowerIndex(input, evaluationColumn);
+    auto secondBound = upperIndex(input, evaluationColumn);
+    if (firstBound > secondBound) {
+      return input;  // all blocks are relevant
+    }
+    // Extract the values from the respective ranges defined by the bounding
+    // indices from vector input.
+    // [input.begin(), firstBound) and [secondBound, input.end()]
+    std::vector<BlockMetadata> vecNe;
+    vecNe.reserve(std::distance(input.begin(), firstBound) +
+                  std::distance(secondBound, input.end()));
+    vecNe.insert(vecNe.end(), input.begin(), firstBound);
+    vecNe.insert(vecNe.end(), secondBound, input.end());
+    return vecNe;
+  }
+};
+
+// SECTION LOGICAL OPERATIONS
+//______________________________________________________________________________
+template <LogicalOperators Operation>
+std::vector<BlockMetadata> LogicalExpressions<Operation>::evaluateOr(
+    std::vector<BlockMetadata>& input, size_t evaluationColumn)
+    requires(Operation == LogicalOperators::OR) {
+  auto relevantBlocksChild1 = child1_->evaluate(input, evaluationColumn);
+  auto relevantBlocksChild2 = child2_->evaluate(input, evaluationColumn);
+  // The result vector, reserve space for the logical upper bound (num blocks)
+  // |relevantBlocksChild1| + |relevantBlocksChild2| regarding the OR
+  // operation (union of blocks)
+  std::vector<BlockMetadata> unionedRelevantBlocks;
+  unionedRelevantBlocks.reserve(relevantBlocksChild1.size() +
+                                relevantBlocksChild2.size());
+  // lambda function that implements the less-than comparison required for
+  // std::set_union
+  auto lessThan = [evaluationColumn](const BlockMetadata& block1,
+                                     const BlockMetadata& block2) {
+    return getIdFromColumnIndex(block1.lastTriple_, evaluationColumn) <
+           getIdFromColumnIndex(block2.lastTriple_, evaluationColumn);
+  };
+  // Given that we have vectors with sorted (BlockMedata) values, we can
+  // use std::ranges::set_union, thus the complexity is O(n + m).
+  std::ranges::set_union(relevantBlocksChild1, relevantBlocksChild2,
+                         std::back_inserter(unionedRelevantBlocks), lessThan);
```

Comment:
1. Nice, I wasn't aware of `std::set_union` :) work smarter, not harder!
2. You can use the `projection` argument of `std::ranges`. The actual comparison is just `<`, you just have to apply the same projection for the left and right blocks. implement this (simpler) lambda, 
then this becomes
`std::ranges::set_uniton(rel1, rel2, back_inserter(result), std::ranges::less_thant{}, getId, getId);`